### PR TITLE
Entryway: non-transactional createAccount

### PR DESCRIPTION
**Pull request by dholms**
Do network requests outside of the createAccount transaction on entrway. If the requests errors, then attempt to delete/cleanup the partially created entryway account.

### CODE REVIEW

File: packages/pds/src/api/com/atproto/server/createAccount.ts [hunk 1]
```diff
@@ -149,17 +150,21 @@ export default function (server: Server, ctx: AppContext) {
...
+            req.log.error(
+              { didKey: ctx.plcRotationKey.did(), handle },
+              'failed to create did:plc',
+            )
+            throw err
+          }
         }
-      })
+      } catch (err) {
+        await cleanupUncreatedAccount(ctx, did)
```
Comment (devinivy) [inline packages/pds/src/api/com/atproto/server/createAccount.ts:165]: I don't think there's an issue here, but we should just ensure there's no way to trigger a pre-existing did from getting "cleaned-up."

File: packages/pds/src/api/com/atproto/server/createAccount.ts [hunk 2]
```diff
@@ -132,6 +120,10 @@ export default function (server: Server, ctx: AppContext) {
             .execute()
         }
 
+        if (!entrywayAssignedPds) {
+          await repoTxn.createRepo(did, [], now)
+        }
```
Comment (devinivy) [inline packages/pds/src/api/com/atproto/server/createAccount.ts:125]: I'm not sure how important this case actually is for the entryway, which shouldn't end-up hosting any repositories.  But if we create a repository and sequence them onto the stream prior to submitting their plc op, it's possible their first event wont get processed by a consumer.

**Review by devinivy — APPROVED** [review]
A couple notes to maybe give a think, but it looks nice to me 👍

File: packages/pds/src/api/com/atproto/server/createAccount.ts (hunk 2 above)
Comment (dholms) [reply packages/pds/src/api/com/atproto/server/createAccount.ts:125]: yup i thought about that

but i also think we just want to remove all this repo logic out of the entryway anyway & it's not currently in use so we don't really run a risk with it

File: packages/pds/src/api/com/atproto/server/createAccount.ts (hunk 1 above)
Comment (dholms) [reply packages/pds/src/api/com/atproto/server/createAccount.ts:165]: good call & we def want to be safe about that. Double checked the logic & the main transaction will throw an error if the did is already registered

**Pull request merged**